source/host_resolver.c

```diff
@@ -558,13 +691,58 @@ static void resolver_thread_fn(void *arg) {
             aws_mem_release(host_entry->allocator, pending_callback);
         }
 
-        aws_mutex_lock(&host_entry->semaphore_mutex);
+        aws_mutex_lock(&host_entry->entry_lock);
+
+        ++host_entry->resolves_since_last_request;
 
         /* we don't actually care about spurious wakeups here. */
-        aws_condition_variable_wait_for(
-            &host_entry->resolver_thread_semaphore, &host_entry->semaphore_mutex, host_entry->resolve_frequency_ns);
+        aws_condition_variable_wait_for_pred(
```

Comment:
delete this comment.
It was bragging about how it wasn't using a predicate.
But now we're using one